apps/api/src/donations/events/stripe-payment.service.ts

```diff
@@ -110,6 +114,157 @@ export class StripePaymentService {
 
     const billingData = getPaymentData(paymentIntent)
 
-    await this.campaignService.donateToCampaign(campaign, billingData)
+    this.campaignService.donateToCampaign(campaign, billingData).then(() => {
+      this.checkForCompletedCampaign(metadata.campaignId as string)
+    })
+  }
+
+  @StripeWebhookHandler('customer.subscription.created')
+  async handleSubscriptionCreated(event: Stripe.Event) {
+    const subscription: Stripe.Subscription = event.data.object as Stripe.Subscription
+    const recurringDonation = await this.recurringDonationService.findSubscriptionByExtId(subscription.id)
+    if (recurringDonation) {
+      //we already have this subscription in our database
+      //stripe already sent us that event
+      return
```

Comment:
Please at least add log for this scenario since it will be hard to track later on if we want to check why we're not handling this subscription event.

Reply:
Sure - will do.